packet_filter.py

```diff
@@ -64,15 +64,21 @@ def detect_tls_events(input_stream):
 
 
 def warn_about_ip_address(ip_address, bad_ips):
-    if ip_address in bad_ips:
-        ioc_source = bad_ips[ip_address]
-        logging.info('CONNECTING WITH BAD IP %s (found in %s)',
-                     ip_address, ioc_source)
+  ip_address = str(ip_address)
+  if ip_address in bad_ips:
+    ioc_sources = bad_ips[ip_address]
+    logging.info('CONNECTING WITH BAD IP %s (found in %s)',
```

Comment:
We should decouple matching and output.
Return matched rules and let the caller decide what to do.

Reply:
See above comment.  I agree, but I wanted to explore the various IOC formats and semantics so as to make informed decisions about what a RuleSet type would look like and how callers would interact with it.  Now I have a much clearer idea of that and an appropriate abstraction designed.  I've documented a sketch of it in that IOCs doc I shared and already plan on sending a PR with that.  The design maps well to Go style, too, so it won't need a redesign when translating.

Reply:
But the caller should also log the indicator, in this case the IP address, and depending on the rule could be multiple signals.  I can change the return value here but I am leaning toward something that looks more like an object (or, likely, a JSON representation of the MISP Event type, much like their .eve file format) in the longer term.